### ISSUE TITLE
Is this real?

### ISSUE THREAD

**Issue by bowbahdoe** [golang-jvm/golang-jvm#2]
I've been looking through the code, but all I can see is the code for JRuby.

Is this a...joke? Am I missing something?

**Comment by szsascha** [comment]
I'm pretty sure the sources are just jruby and its just a bad joke.

**Comment by bowbahdoe** [comment]
Unfortunate - a go -> JVM bytecode compiler would actually be pretty interesting.

**Comment by szsascha** [comment]
Yes. And as funny as it sounds - even useful. Especially for platforms like IBM i or legacy systems in order to migrate to a new platform.

It will definitely has its use cases. The question is just if the work worth it.

**Comment by szsascha** [comment]
It shouldn't even be so hard since go have a really simple syntax. 
The more I think about it, the more I like the challenge. Even because I'm interested in building compilers since a long time but never had the time and a real project to look into it. I think this could be perfect. Just to create a simple proof of concept. (This will not contain goroutines, pointers and similar stuff)

Maybe I'll tackle this task. But I don't know if I can find the time for that. Let's see the coming weeks...

**Comment by bowbahdoe** [comment]
@szsascha Well if you do, feel free to reach out. Recent JVM improvements (virtual threads especially) should make it more viable than ever.

**Comment by szsascha** [comment]
@bowbahdoe I've just found an older project: https://github.com/thomasmodeneis/jgo

**Comment by Thrameos** [comment]
Any progress on there actually being a real copy of this.  I am the author of JPype and I was looking to set up a bridge from Python to Go.  A jvm based go runtime would make this easy.

**Comment by bowbahdoe** [comment]
@Thrameos One thing I did find was that there is a company that has a proprietary JVM go runtime

https://www.remobjects.com/elements/gold/

I think that if we wanted an open source equivalent of that it would be best to wait until value objects land in the JVM. Once that happens it jumps back on my personal list of potential side projects.

**Comment by Thrameos** [comment]
@bowbahdoe  I think that I will go with a different solution then.   

There are two interfaces to Webassembly from Java that could potentially communicate with Go language.   Assuming they have reflection then it would be easy enough to export the modules, types and functions into Java land and no need to rewrite anything in Java.   Just a bit of bridge code and the webassemby compiler for Go.

https://www.graalvm.org/latest/reference-manual/wasm/
https://github.com/wasmerio/wasmer-java

Does that sound useful to anyone?

**Comment by bowbahdoe** [comment]
@Thrameos Maybe a bit of a tangent, but why would I bother with JPype over libpython-clj's java api (as a java consumer) - the zero copy paths for numpy arrays are a hard sell to give up.

**Comment by Thrameos** [comment]
Not really sure what the difference between JPype and libpython-clj in terms of numpy would be.   JPype can directly pass Java arrays to numpy.  The only limitations are those imposed by JNI which only allows certain levels of mapping.   JPype is for Python users to use Java packages as Python modules so it is geared towards that community with "transparent" API (one doesn't need to know Java to use it.)   I am not sure  libpython-clj is in quite the same playing field though it is definitely adjacent.

I am not sure I see the 1 to 1 equivalence between...
https://clj-python.github.io/libpython-clj/embedded.html
https://jpype.readthedocs.io/en/latest/userguide.html

**Comment by headius** [comment]
JRuby guy here. Funny joke! But I actually want to make this happen. Get in touch with me, anyone who's interested.

It's time for JVM Go. We'll call it "Joe".

**Comment by bowbahdoe** [comment]
@headius loop me in - ethan@mccue.dev

**Comment by bowbahdoe** [comment]
<img width="646" height="859" alt="Image" src="https://github.com/user-attachments/assets/6f74151c-2a98-4725-b7c4-8e10b6ccdf9d" />

@headius A bit of a rush job, but...

**Comment by headius** [comment]
@bowbahdoe Beautiful. 😍

I'll see about setting up a shared location where we can all start chatting about this.

**Comment by headius** [comment]
I don't know if any of the original creators of this organization are still around, but I'd love to repurpose it for "Joe".

For now I've started a repository for discussions here: https://github.com/headius/joe/discussions/1
